src/app/clusters/occupancy-sensor-server/OccupancySensingCluster.cpp

```diff
@@ -0,0 +1,342 @@
+}
+
+DataModel::ActionReturnStatus OccupancySensingCluster::ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                                     AttributeValueEncoder & encoder)
+{
+    switch (request.path.mAttributeId)
+    {
+    case Attributes::FeatureMap::Id:
+        return encoder.Encode(mFeatureMap);
+    case Attributes::ClusterRevision::Id:
+        return encoder.Encode(OccupancySensing::kRevision);
+    case Attributes::Occupancy::Id:
+        return encoder.Encode(mOccupancy);
+    case Attributes::OccupancySensorType::Id:
+        return encoder.Encode(FeaturesToOccupancySensorType(mFeatureMap));
+    case Attributes::OccupancySensorTypeBitmap::Id:
+        return encoder.Encode(FeaturesToOccupancySensorTypeBitmap(mFeatureMap));
+    case Attributes::HoldTime::Id:
+    case Attributes::PIROccupiedToUnoccupiedDelay::Id:
+    case Attributes::UltrasonicOccupiedToUnoccupiedDelay::Id:
+    case Attributes::PhysicalContactOccupiedToUnoccupiedDelay::Id:
+        return encoder.Encode(mHoldTime);
+    case Attributes::HoldTimeLimits::Id:
+        return encoder.Encode(mHoldTimeLimits);
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
+    }
+}
+
+DataModel::ActionReturnStatus OccupancySensingCluster::WriteAttribute(const DataModel::WriteAttributeRequest & request,
+                                                                      AttributeValueDecoder & aDecoder)
+{
+    VerifyOrDie(request.path.mClusterId == OccupancySensing::Id);
+
+    switch (request.path.mAttributeId)
+    {
+    case Attributes::HoldTime::Id:
+    case Attributes::PIROccupiedToUnoccupiedDelay::Id:
+    case Attributes::UltrasonicOccupiedToUnoccupiedDelay::Id:
+    case Attributes::PhysicalContactOccupiedToUnoccupiedDelay::Id: {
+        if (!mHasHoldTime)
+        {
+            return Protocols::InteractionModel::Status::UnsupportedAttribute;
+        }
+        uint16_t newHoldTime;
+        ReturnErrorOnFailure(aDecoder.Decode(newHoldTime));
+        return NotifyAttributeChangedIfSuccess(request.path.mAttributeId, SetHoldTime(newHoldTime));
+    }
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
+    }
+}
+
+CHIP_ERROR OccupancySensingCluster::Attributes(const ConcreteClusterPath & clusterPath,
+                                               ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder)
+{
+    AttributeListBuilder listBuilder(builder);
+
+    const AttributeListBuilder::OptionalAttributeEntry optionalAttributes[] = {
+        { mHasHoldTime, Attributes::HoldTime::kMetadataEntry },
+        { mHasHoldTime, Attributes::HoldTimeLimits::kMetadataEntry },
+        { mHasHoldTime && mFeatureMap.Has(Feature::kPassiveInfrared), Attributes::PIROccupiedToUnoccupiedDelay::kMetadataEntry },
+        { mHasHoldTime && mFeatureMap.Has(Feature::kUltrasonic), Attributes::UltrasonicOccupiedToUnoccupiedDelay::kMetadataEntry },
```

Comment:
There should be an option to decide whether to show the previous delay attributes or not... They are deprecated, but can/should be enabled by default for legacy compat.